MetaDataset.py

```diff
@@ -1017,10 +1107,13 @@ def is_less_than_num_seqs(self, n):
     :param int n:
     :rtype: bool
     """
-    if n < len(self.dataset_sorted_seq_idx_list):
-      return True
+    if self._can_use_num_seqs:
+      return n < self._num_seqs
```

Comment:
Why do you need this? Wouldn't the other code already cover all cases?

Reply:
Same as above.

Reply:
I don't understand. You are anyway using `dataset_sorted_seq_idx_list` in all cases? So this should work?

Note that `is_less_than_num_seqs` is a function like `get_data`, `get_sequence_length` and many others: This is used while iterating through the data, and supposed to work only for current loaded seqs, or maybe the next following seq. This is not supposed to be used for other seqs. See e.g. `CachedDataset2.is_less_than_num_seqs`.

Reply:
´n < self.dataset_sorted_seq_idx_list` also works for the known num_seqs case, yes. But if n is larger we don't want to call `_expand_dataset_sec_idxs()`. Or at least not without rewriting it. But I think it should not be called in that case because in that mode of operation we never expand the list, so checking at epoch end whether we have to is misleading.

Reply:
I could check manually that `is_less_than_num_seqs` is false for all sub-dataset, but then we're back at our discussion about num_seqs. If the sequence order is known in advance, than `is_less_than_num_seqs` <=> n < len(seq_order). So I could write ìf self._num_seqs is not None` instead of `if self._can_use_num_seqs:` if I wanted to avoid using the latter.

Reply:
Why is `dataset_sorted_seq_idx_list` not available? As I said, it's bad to have separate logic if this is not really needed. Why can't `_expand_dataset_sec_idxs` just handle both cases? It should be simple to define `dataset_sorted_seq_idx_list` for the case when `num_seqs` is available, or not?